src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -539,6 +539,28 @@ private static X509ChainStatusFlags MapVerifyErrorToChainStatus(Interop.Crypto.X
                         }
                     }
                 }
+
+                var candidatesByReference = new HashSet<X509Certificate2>(
+                    candidates,
+                    ReferenceEqualityComparer<X509Certificate2>.Instance);
+
+                // Dispose any certificates we cloned in, but didn't end up needing.
+                // Since extraStore was provided by users, don't dispose anything it contains.
+                Debug.Assert(storesToCheck.Length > 0, "storesToCheck.Length > 0");
+                Debug.Assert(storesToCheck[0] == extraStore, "storesToCheck[0] == extraStore");
+
+                for (int i = 1; i < storesToCheck.Length; i++)
```

Comment:
We're skipping the first one because all of its certs were used?  Is it possible there were dups within the store?